source/librii/u8/U8.cpp

```diff
@@ -51,41 +51,42 @@ struct LowU8Archive {
 #define rvlArchiveNodeIsFolder(node) ((node).packed_type_name & 0xff000000)
 #define rvlArchiveNodeGetName(node) ((node).packed_type_name & 0x00ffffff)
 
-static inline const rvlArchiveNode*
+static inline Result<const rvlArchiveNode*>
 rvlArchiveHeaderGetNodes(const rvlArchiveHeader* self) {
-  assert(self->nodes.offset >= sizeof(rvlArchiveHeader));
+  EXPECT(self->nodes.offset >= sizeof(rvlArchiveHeader), "Invalid nodes");
   return (const rvlArchiveNode*)(((u8*)self) + self->nodes.offset);
 }
-static inline const u8*
+static inline Result<const u8*>
 rvlArchiveHeaderGetFileData(const rvlArchiveHeader* self) {
-  assert(self->files.offset >= sizeof(rvlArchiveHeader));
+  EXPECT(self->files.offset >= sizeof(rvlArchiveHeader), "Invalid file data buffer");
   return (const u8*)((u8*)self + self->files.offset);
 }
 
-template <typename T> bool SafeMemCopy(T& dest, std::span<const u8> data) {
-  assert(data.size_bytes() >= sizeof(T));
-  if (data.size_bytes() < sizeof(T)) {
-    std::fill((u8*)&dest, (u8*)(&dest + 1), 0);
-    return false;
-  }
-
+template <typename T>
+static Result<void> SafeMemCopy(T& dest, rsl::byte_view data,
+                                std::string fail_msg) {
+  EXPECT(data.size_bytes() >= sizeof(T));
   std::memcpy(&dest, data.data(), sizeof(T));
-  return true;
 }
 
-bool RangeContains(std::span<const u8> range, const void* ptr) {
+static bool RangeContains(rsl::byte_view range, const void* ptr) {
   return ptr >= range.data() && ptr < (range.data() + range.size());
 }
-bool RangeContainsInclusive(std::span<const u8> range, const void* ptr) {
+static bool RangeContainsInclusive(rsl::byte_view range, const void* ptr) {
   return ptr >= range.data() && ptr <= range.data() + range.size();
 }
 
-Result<void> LoadU8Archive(LowU8Archive& result, std::span<const u8> data) {
-  if (!SafeMemCopy(result.header, data))
-    return std::unexpected("Invalid header");
+Result<bool> IsDataU8Archive(rsl::byte_view data) {
```

Comment:
I'd rather it stay `bool`, since it doesn't *really* matter to anyone if it's "really really not a U8 file", just serving to complicate the API.